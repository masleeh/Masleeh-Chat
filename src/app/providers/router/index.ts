import AppRouter from "./ui/AppRouter";
import { AppRouterProvider, router as AppBrowserRouter } from "./ui/RouterProvider";

export {
    AppRouter,
    AppRouterProvider,
    AppBrowserRouter
}