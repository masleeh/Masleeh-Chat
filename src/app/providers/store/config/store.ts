import { ReducersMapObject, configureStore } from "@reduxjs/toolkit"
import { IStateSchema } from "./StateSchema"
import { userReducer } from "entities/User"


export const createReduxStore = (initialState?: IStateSchema) => {
    const rootReducers: ReducersMapObject<IStateSchema> = {
        user: userReducer
    }

    return configureStore<IStateSchema>({
        reducer: rootReducers,
        devTools: true,
        preloadedState: initialState
    })
}

export type RootState = ReturnType<typeof createReduxStore>['getState']
export type AppDispatch = ReturnType<typeof createReduxStore>['dispatch']