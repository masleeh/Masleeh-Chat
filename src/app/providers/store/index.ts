import StoreProvider from "./StoreProvider";
import type { IStateSchema } from "./config/StateSchema";

export {
    StoreProvider
}

export type {
    IStateSchema
}