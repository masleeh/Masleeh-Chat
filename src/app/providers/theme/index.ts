import AppThemeProvider from './ui/ThemeProvider'

export {
    AppThemeProvider
}