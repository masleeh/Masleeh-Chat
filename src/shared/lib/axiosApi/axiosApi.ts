import axios from 'axios'

const $api = axios.create({
    withCredentials: true,
    baseURL: import.meta.env.VITE_API_URL
})

$api.interceptors.request.use((config) => {
    config.headers.Authorization = `Bearer ${localStorage.getItem(import.meta.env.VITE_LOCALSTORAGE_TOKEN_KEY)}`
    return config
})

$api.interceptors.response.use((config) => {
    return config   
}, async (error) => {
    const originalRequest = error.config;
    if (error.response.status === 401 && error.config && !error.config._isRetry) {
        originalRequest._isRetry = true
        try {
            const response = await axios.get(`${import.meta.env.VITE_API_URL}/auth/refresh`, { withCredentials: true })
            localStorage.setItem(import.meta.env.VITE_LOCALSTORAGE_TOKEN_KEY, response.data.access_token)
            return $api.request(originalRequest)
        } catch (error) {
            console.log("Not authorized")
        }
    }
    throw error
})

export default $api