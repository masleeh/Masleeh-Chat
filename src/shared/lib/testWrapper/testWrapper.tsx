import { render } from "@testing-library/react"
import { IStateSchema, TestStoreProvider } from "app/providers/store";
import i18nTest from 'shared/config/i18n/i18nTest'
import { I18nextProvider } from "react-i18next"
import { MemoryRouter } from "react-router-dom";
import { DeepPartial, EnhancedStore } from "@reduxjs/toolkit";
import { AppThemeProvider } from "app/providers/theme";

export interface componentRenderOptions {
    route?: string;
    initialState?: DeepPartial<IStateSchema>;
    store?: EnhancedStore
}

const testWrapper = (component: React.ReactNode, options: componentRenderOptions = {}) => {
    const {
        route = '/',
        initialState,
        store
    } = options

    return render(
        <TestStoreProvider initialState={initialState} store={store}>
            <AppThemeProvider>
                <MemoryRouter initialEntries={[route]}>
                    <I18nextProvider i18n={i18nTest}>
                        {component}
                    </I18nextProvider>
                </MemoryRouter>
            </AppThemeProvider>
        </TestStoreProvider>
    )
}

export default testWrapper