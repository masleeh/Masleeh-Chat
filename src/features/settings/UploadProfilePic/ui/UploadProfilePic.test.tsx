import { DeepPartial } from '@reduxjs/toolkit'
import {screen} from '@testing-library/react'
import { IStateSchema, createReduxStore } from 'app/providers/store'
import $api from 'shared/lib/axiosApi/axiosApi'
import testWrapper from "shared/lib/testWrapper/testWrapper"
import { UploadProfilePic } from '..'
import userEvent from '@testing-library/user-event'

const axiosMock = vi.spyOn($api, 'post')
const state: DeepPartial<IStateSchema> = {
    user: {
        userData: {
            username: 'masleeh',
            user_id: 'fowe8vke7c;wh',
            profile_pic: 'http://localhost:5000/users/Ea3bptxnJF2-dqp6Wk8SHEZOlVkw_oyaJz4QlO6h.jpg'
        }
    }
}

const values = ['masleeh']
const str = JSON.stringify(values);
const blob = new Blob([str]);
const file = new File([blob], 'values.json', {
    type: 'application/JSON',
});
File.prototype.text = vi.fn().mockResolvedValueOnce(str)

describe('UploadProfilePic component', () => {
    it('Check UploadProfilePic', () => {
        const component = testWrapper(<UploadProfilePic />)
        expect(component).toMatchSnapshot()
    })

    it('Check functional', async () => {
        axiosMock.mockResolvedValue({
            profile_pic: 'http://localhost:5000/static/j437rUnzlSQucxMl2xT1dVU05BRP2IOsBU1IAROs.jpg'
        })

        const store = createReduxStore(state as IStateSchema)
        testWrapper(<UploadProfilePic />, {store: store})

        const input = screen.getByLabelText('upload profile pic input')
        await userEvent.upload(input, file)

        setTimeout(() => {
            const newState = store.getState()
            expect(newState.user.userData?.profile_pic)
                .toEqual('http://localhost:5000/static/j437rUnzlSQucxMl2xT1dVU05BRP2IOsBU1IAROs.jpg')
        }, 1500)
    })

})