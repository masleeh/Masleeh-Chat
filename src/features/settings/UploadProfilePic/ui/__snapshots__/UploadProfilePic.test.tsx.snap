// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UploadProfilePic component > Check UploadProfilePic 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloudUploadIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
            />
          </svg>
        </span>
        upload_pic_button
        <input
          accept="image/jpeg"
          aria-label="upload profile pic input"
          class="css-1bigxrc"
          type="file"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CloudUploadIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
          />
        </svg>
      </span>
      upload_pic_button
      <input
        accept="image/jpeg"
        aria-label="upload profile pic input"
        class="css-1bigxrc"
        type="file"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
